pkg/controller/controller.go

```diff
@@ -409,6 +425,24 @@ func (c *Controller) runAction(ctx context.Context, as *crv1alpha1.ActionSet, aI
 			}
 			var rf func(*crv1alpha1.ActionSet) error
 			if err != nil {
+				// If error is because of context canceled && timeout then update the actionset with failed status 
+				if ctx.Err() != nil {
+					bpName := as.Spec.Actions[aIDX].Blueprint
+					bp, _ := c.crClient.CrV1alpha1().Blueprints(as.GetNamespace()).Get(context.Background(), bpName, v1.GetOptions{})
```

Comment:
Why are we using `context.Background()` here? Should we use `ctx` instead?
Same for other occurrences in the same function

Reply:
@PrasadG193 I believe If we use `ctx` that will give an error since `ctx` is already canceled at this point